lexers/org.lua

```diff
@@ -0,0 +1,137 @@
+--[[ Textadept Org module -- Org agenda LPeg lexer
+Copyright (c) 2012 joten
+Copyright (c) 2025 Matěj Cepl (@mcepl everywhere)
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+--]]
+
+local lexer = lexer
+local word_match = lexer.word_match
+local P, R, S, V = lpeg.P, lpeg.R, lpeg.S, lpeg.V
+lpeg.locale(lpeg)
+
+local lex = lexer.new(...)
+
+--[[Overview. Used examples.
+* Heading 1-5 (8)
+	TODO special color, bold
+	DONE special color, bold
+	[#priority]
+	:tag: bold
+<date> / [date] like heading 3, underlined
+	keywords: CLOSED: DEADLINE: SCHEDULED: like heading 3, not underlined
+| table | like heading 1
+[[link][description] ]
+formatting:
+	*bold*
+	/italic/
+	_underline_
+	+strike+
+--]]
+
+-- Font formats.
+lex:add_rule('bold', lex:tag('BOLD',      '*' * lexer.word^1 * '*'))
+lex:add_rule('italic', lex:tag('ITALIC',    '/' * lexer.word^1 * '/'))
+lex:add_rule('underline', lex:tag('UNDERLINE', '_' * lexer.alnum^1 * '_'))
+
+-- ToDos.
+lex:add_rule('todo', lex:tag('TODO', lex:word_match('TODO')))
+lex:set_word_list('TODO', {'TODO', 'DELEGATED', 'WAITING'})
+lex:add_rule('done', lex:tag('DONE', lex:word_match('DONE')))
+lex:set_word_list('DONE', {'DONE'})
+lex:add_rule('wontfix', lex:tag('WONTFIX', lex:word_match('WONTFIX')))
+lex:set_word_list('WONTFIX', {'WONTFIX', 'INVALID'})
+local agenda_tags = P('TODO') + 'DONE' + 'WONTFIX'
+
+-- XXX I don’t care about priority, if anybody cares, send me a patch
+-- Also, I don’t think that the TAG is the right name, just use a custom tag
+-- local priority = lex:tag('PRIORITY', '[#' * S('ABC') * ']')
+-- local tags = lex:tag('TAG', ':' * (lexer.word + lexer.punct)^1 * lexer.newline)
+
+-- XXX not sure what it is good for
+-- Keywords.
+-- lex:add_rule('keyword', lex:tag(lexer.KEYWORD, lex:word_match(lexer.KEYWORD)))
+-- lex:set_word_list(lexer.KEYWORD, {'SCHEDULED', 'DEADLINE', 'CLOSED'})
```

Comment:
This unused code should probably be removed to clean things up.

Reply:
Done